dipy/reconst/dti.py

```diff
@@ -1905,7 +1914,65 @@ def design_matrix(gtab, dtype=None):
     return -B
 
 
-def quantize_evecs(evecs, odf_vertices=None):
+def _quantize_evecs_parallel(data, odf_vertices, v, nbr_processes):
+    if nbr_processes is -1:
+        try:
+            nbr_processes = cpu_count()
+        except NotImplementedError:
+            warn("Cannot determine number of cpus. \
+                 returns quantize_evecs(..., nbr_processes=1).")
+            return quantize_evecs(data, odf_vertices, v, nbr_processes=1)
+
+    data = data[..., :, v]
+    shape = list(data.shape)
+    data = np.reshape(data, (-1, shape[-1]))
+    n = data.shape[0]
+    nbr_chunks = nbr_processes ** 2
+    chunk_size = int(np.ceil(n / nbr_chunks))
+    indices = list(zip(np.arange(0, n, chunk_size),
+                       np.arange(0, n, chunk_size) + chunk_size))
+
+    with InTemporaryDirectory() as tmpdir:
+        data_file_name = path.join(tmpdir, 'data.npy')
+        np.save(data_file_name, data)
+
+        pool = Pool(nbr_processes)
+
+        peaks_res = pool.map(_quantize_evecs_parallel_sub,
+                             zip(repeat(data_file_name),
+                                 indices,
+                                 repeat(odf_vertices),
+                                 repeat(v)))
```

Comment:
Instead of using temporary file, What do you think about using the technique explained [here](http://thousandfold.net/cz/2014/05/01/sharing-numpy-arrays-between-processes-using-multiprocessing-and-ctypes/).

Reply:
OK - I implemented this in https://github.com/nipy/dipy/pull/1164/commits/bf858f9cd018e447f328e85b20cad0135f891a2b